src/main/java/com/eleks/academy/whoami/model/response/HistoryDetails.java

```diff
@@ -0,0 +1,25 @@
+package com.eleks.academy.whoami.model.response;
+
+import com.eleks.academy.whoami.repository.HistoryChat;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Data
+@NoArgsConstructor
+public class HistoryDetails {
+    private Map<List<String>, List<String>> historyMap = new HashMap<>();
+
+    public HistoryDetails(HistoryChat historyChat) {
+        historyMap.put(historyChat.getQuestions(), historyChat.getAnswers());
```

Comment:
That is not good usage of HashMap, you are putting whole list as Map key and whole list as value, so you basically always have a map with one element. So you only getting overhead in memory and speed without any advantage.